src/rust/lib/frp/src/io/keyboard2.rs

```diff
@@ -0,0 +1,258 @@
+#![allow(missing_docs)]
+// FIXME[WD]: THIS FILE IS IN WORK IN PROGRESS STATE. WILL BE FINISHED IN THE NEXT PR.
```

Comment:
If it's not ready yet, why we use it and want to merge it to master?

Reply:
Because I thought that I will be able to finish it during this task, but it appeared bigger than it was. This is my next task, but this code is already integrated with scene, application etc. I can spend time on de-integrating it, but I thought it doesn't matter if my next task is about to finish it.